gammapy/datasets/map.py

```diff
@@ -1114,12 +1113,11 @@ def plot_residuals(
         return ax_spatial, ax_spectral
 
     def stat_sum(self):
-        """Total statistic function value given the current model parameters and priors."""
+        """Total statistic function value given the current model parameters and priors set on the models."""
+        counts, npred = self.counts.data.astype(float), self.npred().data
         prior_stat_sum = 0.0
         if self.models is not None:
-            prior_stat_sum = self.models.parameters.prior_stat_sum()
-
-        counts, npred = self.counts.data.astype(float), self.npred().data
+            prior_stat_sum = prior_fit_statistic(self.models.priors)
```

Comment:
Wouldn't this double count the prior, when it is evaluated again at the `Datasets.stat_sum()` level?